An option to control SSL certificate validation added. (#180)
logstash-plugins/logstash-input-elasticsearch: docs/index.asciidoc

If enabled, SSL will be used when communicating with the Elasticsearch
server (i.e. HTTPS will be used instead of plain HTTP).

[id="plugins-{type}s-{plugin}-ssl_certificate_verification"]
===== `ssl_certificate_verification`

* Value type is <<boolean,boolean>>
* Default value is `true`

> Option to validate the server's certificate. Disabling this severely compromises security.

Reviewer: Although our docs call out _"Disabling this severely compromises security"_ and links to a 12-page white-paper that is tangentially-related to improper certificate validation, I think it may be worth calling out at least a few of the abuse scenarios in the doc here:

```suggestion
Option to validate the server's certificate. Disabling this severely compromises security.
When certificate validation is disabled this plugin implicitly trusts the machine
resolved at the given address without validating its proof-of-identity, which can
lead to this plugin transmitting credentials to or processing data from an untrustworthy
man-in-the-middle or other compromised infrastructure.
```

If enabled, SSL will be used when communicating with the Elasticsearch
server (i.e. HTTPS will be used instead of plain HTTP).

[id="plugins-{type}s-{plugin}-ssl_certificate_verification"]
===== `ssl_certificate_verification`

* Value type is <<boolean,boolean>>
* Default value is `true`

Option to validate the server's certificate. Disabling this severely compromises security.

> When certificate validation is disabled this plugin implicitly trusts the machine
> resolved at the given address without validating its proof-of-identity, which can
> lead to this plugin transmitting credentials to or processing data from an untrustworthy
> man-in-the-middle or other compromised infrastructure.
> For more information on disabling certificate verification please read
> https://www.cs.utexas.edu/~shmat/shmat_ccs12.pdf

Reviewer 2: ```suggestion
When certificate validation is disabled, this plugin implicitly trusts the machine
resolved at the given address without validating its proof-of-identity.
In this scenario, the plugin can transmit credentials to or process data from an untrustworthy
man-in-the-middle or other compromised infrastructure.
More information on the importance of certificate verification:
**https://www.cs.utexas.edu/~shmat/shmat_ccs12.pdf**.
```
